sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricResults.java

```diff
@@ -0,0 +1,34 @@
+package org.apache.beam.sdk.metrics;
+
+import autovalue.shaded.com.google.common.common.collect.Iterables;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+
+/**
+ * Methods for interacting with the metrics of a pipeline that has been executed. Accessed via
+ * {@link PipelineResult#metrics()}.
```

Comment:
I initially had it added to the `PipelineResult` but it seemed nice to group all of the metrics-querying functionality into a single place. This also makes it easier to have default implementations of some of the `Metrics` logic that different `PipelineResult` instances can build on when producing their `Metrics` implementation (for instance, it is a class right now to provide some defaults).